### ISSUE TITLE
New openstreetmap-carto release, v4.5.0

### ISSUE THREAD

**Issue by kocio-pl** [openstreetmap/chef#138]
A new version of openstreetmap-carto, [v4.5.0](https://github.com/gravitystorm/openstreetmap-carto/releases/tag/v4.5.0), has been released.

**Comment by math1985** [comment]
Thanks for updating!

Could it be there is something wrong with the London tile server? See https://github.com/gravitystorm/openstreetmap-carto/issues/2955 .

**Comment by kocio-pl** [comment]
I have the same issue with z5 in Western Europe as reported, I also have similar problem on z7 (for example eastern part of Poland and other eastern countries still show natural reserves and railways).

```
$ host a.tile.openstreetmap.org
a.tile.openstreetmap.org is an alias for tile.geo.openstreetmap.org.
tile.geo.openstreetmap.org is an alias for pl.tile.openstreetmap.org.
pl.tile.openstreetmap.org is an alias for germany.tile.openstreetmap.org.
germany.tile.openstreetmap.org has address 144.76.70.77
germany.tile.openstreetmap.org has address 81.7.11.83
germany.tile.openstreetmap.org has address 185.66.195.245
```

**Comment by tomhughes** [comment]
Telling me what cache you're using is next to useless - if you want to help go to http://tile.openstreetmap.org/cgi-bin/debug and tell me which render server you're using.

**Comment by kocio-pl** [comment]
Render Server: orm.openstreetmap.org

**Comment by math1985** [comment]
I'm using scorch, and this server renders fine.

**Comment by tomhughes** [comment]
Right, and that server hasn't finished the low zoom re-render yet so this is expected.

**Comment by math1985** [comment]
Good to hear! Is there anywhere where we can see if the re-render process is finished?

**Comment by kocio-pl** [comment]
I thought that z0-z12 was rendered in one go, probably at the start of deploying new version of osm-carto - is it different?

**Comment by tomhughes** [comment]
It is rendered in one go yes. That "go" has not yet finished.

**Comment by kocio-pl** [comment]
But it lasts for a few days already with no change, like 5 probably (both on z5 and z7). Once I remember it was like about few hours peak and that was all. That's why I'm curious what's going on.

**Comment by DaveF63** [comment]
For me, the problem is older than 6 days. Weeks probably, As it didn't resolve itself I only posted about it now. 
Do low levels get re-rendered every update, even if the appearance of entities don't change?

**Comment by tomhughes** [comment]
Eh? I assumed the point was that this stylesheet change had added or removed a lot of detail at z5? So I don't see how this can predate this change?

Am I missing something? Are the two versions of z5 we are seeing not the before and after versions for this stylesheet change?

**Comment by kocio-pl** [comment]
This particular rendering artifact is new, of course. But I also remember that with last release there were some quirks with rendering of z0-z12, which was like few days later on some servers than some others: https://github.com/openstreetmap/chef/issues/131#issuecomment-338420147.

**Comment by kocio-pl** [comment]
Looks like orm is rendering low/medium zoom at last (it's not clear what happened earlier as we have missing data for v4.5.0 deployment on 17-19.11): 
https://munin.openstreetmap.org/openstreetmap/orm.openstreetmap/index.html#renderd

I also see my problems described here are gone and USA or China are also clean now on z5.

Te question is however what just happened and if that could be avoided in the future. Tom, what is the algorithm/rule for rerendering low and medium zoom levels on the OSMF servers?

**Comment by tomhughes** [comment]
The short version is that `render_old` had hung and I gave it a kick.

That's not unusual - it doesn't seem to be the most stable of things. I don't have any good ideas how to work around it though - we already restart it if it exits with an error which copes with the case where renderd is restarted and it bombs out but in this case it had seemingly just hung.

**Comment by kocio-pl** [comment]
Thanks, Tom! So should we just write here in case something seems to be stuck next time?

I'm also interested in the schedule - and if it's fired automatically or manually?

**Comment by gravitystorm** [comment]
The updates to the tileservers are handled via chef. All the configuration is available in the chef repo, although I realise it can be a bit intimidating to read, it's understandable particularly if you're used to how tileservers work.

https://github.com/openstreetmap/chef/blob/58b212656b189b32103e25209df92c02b4535420/cookbooks/tile/recipes/default.rb#L329-L338

The above link shows the heart of the update process. It subscribes for notifications that the ["git style_directory"](https://github.com/openstreetmap/chef/blob/0993dbc450a5678cfcea86dd71cf5873b18d86f3/cookbooks/tile/recipes/default.rb#L329-L335) resource has run, which happens when the [config is updated](https://github.com/openstreetmap/chef/commit/e5b25cd4c7f738ffd417a631999834d70ac552f6) for a new release. The `execute` resource that I linked to then runs the `carto` compiler, and issues two notifications - one to restart renderd, and the other to trigger the restarting of the update-lowzoom service. That service definition is [here](https://github.com/openstreetmap/chef/blob/58b212656b189b32103e25209df92c02b4535420/cookbooks/tile/templates/default/update-lowzoom.erb)

So in short, it triggers automatically when the style gets updated.

In theory, you can use the chef repo to set up a perfect clone of the production tileservers, so that you can tinker with it and see how it works locally. We've set up [a few of the cookbooks](https://github.com/openstreetmap/chef/blob/58b212656b189b32103e25209df92c02b4535420/.kitchen.yml) to work with test-kitchen, but not all of them yet. Unfortunately the `tile` cookbook isn't one of the ones that currently works with test-kitchen but hopefully someone can get that sorted.

If anyone needs a more thorough explanation of how the chef cookbooks work I'm happy to help. I'm not actually familiar with the OSMF tileserver setup but I'm experienced enough at reading Chef cookbooks to figure it out :-)

**Comment by tomhughes** [comment]
To follow up a little more I should probably explain how the `update-lowzoom` service works...

Firstly as @gravitystorm says by the time it starts the style has been updated and renderd has been restart so any new tiles rendered will be in the new style from that point on.

What `update-lowzoom` does is to run `render_old` in a loop until it completes, using the timestamp of the new style as the reference time so that any z0-12 tiles older than that will be rerendered. In particular note that this ensures that if it does get restarted it will quickly skip any done by a previous pass.

Once `render_old` has completed all z0-12 tiles the `planet file` is touched with the time of the style file meaning that all high zooms tiles older than the style will now be rerendered if possible when requested.

What I did on orm yesterday was just to kill `render_old` so that the loop would restart it and finish off the rest of the low zoom tiles.

There is a separate script called `render-lowzoom` which runs on the first sunday of each month and rerenders the low zoom tiles in much the same way.

**Comment by kocio-pl** [comment]
Thanks, that explains a lot! 

I've made a link on osm-carto Wiki page, but I think that you could probably put these things in a place that is more visible for the rest of OSM community than a closed ticket in some obscure subproject :smile: (maybe https://blog.openstreetmap.org/category/operations/ would do? ). 

This might look like a boring nerdy stuff, but for me it's a precious knowledge and I think this is a great way to make OWG more open and approachable. I remember, Andy, that you wanted to [invite more people to work there](https://blog.gravitystorm.co.uk/2016/07/28/getting-involved-in-owg/), and I guess it could really work this way.
